FSA2021-167: Add checkmark contrast background

rise-erpelding opened the pull request:
# Description
Previously, we used white-colored checkmarks to tell the user that they have placed a tile in the correct position. But what if the tile is white or light-colored? To avoid this problem, each check mark now has a small background that contrasts with the checkmark--the background is white and the checkmark is now black.

Since the tiles can vary greatly in size depending on the user's chosen grid size or screen size, the size of the checkmark (and its background) now depend on the size of the tile instead of remaining a static value.

## Spec

See [FSA2021-167](https://sparkbox.atlassian.net/browse/FSA2021-167)

## To Validate
1. Pull down the branch and install dependencies.
2. Run `npm start` and navigate to http://localhost:1234
3. Generate a grid of size 3 and press the 'play game with this grid' (you'll need to scroll down to see it) button.
4. View the checkmarks on the correct tiles (which should at a minimum be in the 4 corners), ensure that they're positioned and sized correctly.
5. Generate a grid of size 16 and repeat step 4.

rise-erpelding committed:
feat: add circle background to checkbox

-Change color of checkbox from white to black
-Add white circle background to checkbox
-Changed font-size from default to 0.75rem to work in smaller tiles

FSA2021-167

rise-erpelding committed:
refactor: remove use of padding for positioning

-The checkmark is horizontally positioned by
centering it
-The checkmark is vertically positioned with line-height
-Instead of padding, the circle shape is created by setting an explicit
height and width

FSA2021-167

rise-erpelding committed:
refactor: make checkmark size dynamic

-Sets checkmark and checkmark circle size using the --grid variable
(which tells us how many squares across our grid is) as a reference.
This allows the checkmark and checkmark circle to be sized in proportion
to the size of the color tile.
Now:
  -Smaller tiles get smaller checkmarks (which are not necessarily
  legible, but as they're meant to exist to tell the user they've put a
  tile in the right spot, they are visible)
  -Larger tiles get larger checkmarks
  -Size of the checkmark and circle are restricted so that they are not
  too small and more importantly, not too big

  FSA2021-167

rise-erpelding committed:
feat: restrict size of grid to between 3 and 16

-A grid size of 2 is essentially just the corner colors that the user
picked, which is no fun and definitely serves no purpose when playing a
game. Therefore, 3 is now the minimum.
-A grid size of 20 is a little crazy for the user and poses some UI
challenges. Therefore, it is now restricted to 16 (which doesn't seem
like a lot but is actually 144 fewer tiles to deal with).

FSA2021-167

kaseybon commented on src/sass/components/_color-grid.scss:
```diff
@@ -48,10 +48,24 @@
-        padding: 0.35rem;
-        color: white;
+          content: "\2713"; // checkmark
+          --checkmark-circle-size: calc(var(--grid) / 5);
```
It would be a good idea to move all of your custom properties into the same settings file. While trying to understand the numbers being used I found myself having to find where `--grid` was defined and then that included another property and I had to track it down. Getting everything in one place will make it easier to understand how the different variables work together.

Grouping them all together will also make it so you don't have to remember which properties exist. If you come back to the project in a couple of months you probably won't remember what all properties are available and you could find yourself creating duplicate properties.

rise-erpelding committed:
fix: update scope of the --size variable

-Previously, the `--size` CSS variable (custom property) was scoped to a
single class. To avoid any issues with variables not being available to
all elements, the JavaScript now adds the CSS property to the root
instead of that element.
-Additionally adds a helper function to abstract some of that code from
the fillColors function.
-Adds a blue variable as a debug

FSA2021-167

rise-erpelding committed:
notes from pairing with kasey and zastrow

rise-erpelding committed:
cleaning up
